src/codegen.cpp

```diff
@@ -4251,7 +4283,8 @@ static jl_cgval_t emit_invoke(jl_codectx_t &ctx, const jl_cgval_t &lival, const
                     bool cache_valid = ctx.use_cache;
                     bool external = false;
                     if (ctx.external_linkage) {
-                       if (0 && jl_object_in_image((jl_value_t*)codeinst)) {
+                       if (specsig && jl_object_in_image((jl_value_t*)codeinst)) {
```

Comment:
In offline discussion @vtjnash and I decided that this needs to do an additional check to see if the fptr is actually from the pkgimage in which the CodeInstance is defined.